src/main/java/org/spongepowered/common/mixin/api/mcp/village/MerchantOfferMixin_API.java

```diff
@@ -0,0 +1,114 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.api.mcp.village;
+
+import net.minecraft.item.MerchantOffer;
+import org.spongepowered.api.data.persistence.DataContainer;
+import org.spongepowered.api.data.persistence.Queries;
+import org.spongepowered.api.item.inventory.ItemStack;
+import org.spongepowered.api.item.inventory.ItemStackSnapshot;
+import org.spongepowered.api.item.merchant.TradeOffer;
+import org.spongepowered.api.util.annotation.NonnullByDefault;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.common.util.Constants;
+
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+@NonnullByDefault
+@Mixin(MerchantOffer.class)
+public abstract class MerchantOfferMixin_API implements TradeOffer {
+
+    @Shadow public abstract net.minecraft.item.ItemStack getBuyingStackFirst();
+    @Shadow @Nullable public abstract net.minecraft.item.ItemStack getBuyingStackSecond();
+    @Shadow public abstract net.minecraft.item.ItemStack getSellingStack();
+    @Shadow public abstract int shadow$getUses();
+    @Shadow public abstract int func_222214_i(); // getMaxUses
+    @Shadow public abstract boolean hasNoUsesLeft();
+    @Shadow public abstract boolean getDoesRewardExp();
+
+    @Override
+    public ItemStackSnapshot getFirstBuyingItem() {
+        return ((ItemStack) this.getBuyingStackFirst()).createSnapshot();
+    }
+
+    @Override
+    public boolean hasSecondItem() {
+        return this.getBuyingStackSecond() != net.minecraft.item.ItemStack.EMPTY;
+    }
+
+    @Override
+    public Optional<ItemStackSnapshot> getSecondBuyingItem() {
+        if (this.getBuyingStackSecond() == null) {
+            return Optional.empty();
+        }
+        return Optional.of(((ItemStack) this.getBuyingStackSecond()).createSnapshot());
+    }
+
+    @Override
+    public ItemStackSnapshot getSellingItem() {
+        return ((ItemStack) this.getSellingStack()).createSnapshot();
+    }
+
+    @Override
+    public int getUses() {
```

Comment:
This would have to be intrinsic because it's soft shadowing a target method as well.